src/modules/item/reducer.ts

```diff
@@ -127,7 +145,35 @@ export function itemReducer(state: ItemState = INITIAL_STATE, action: ItemReduce
         ...state,
         data: {
           ...state.data,
-          [item.id]: { ...item }
+          ...toItemObject([item])
+        },
+        loading: loadingReducer(state.loading, action),
+        error: null
+      }
+    }
+    case FETCH_COLLECTIONS_SUCCESS: {
+      const { collections } = action.payload
+      const items = ([] as Item[]).concat.apply(
```

Comment:
what do you think about 

```ts
collections.reduce<Item>((items, collection) => items.concat(collection.items), [])
``` 

?

Reply:
Much better, I let my inner js ninja come out, I'm sorry